### PR TITLE
CU_#860qewjwn SEO contents improved

### PR DESCRIPTION
**Pull request by Ayomikuolatunji**

### CODE REVIEW

File: src/generators/DetailsPageGenerator.ts [hunk 1]
```diff
@@ -6,21 +6,31 @@ export const spaceToDashes = (name: string) => {
...
-  return `Nutrient facts from USDA and NIH with the option to filter by nutrients, age and gender for 100gm of  ${foodName} in ${category} category.`;
-};
-
 const path = require("path");
-
 const ff_nutrition_facts = require("../../data/foundation_food_nutrition_facts.json");
 const sr_legacy_nutrition_facts = require("../../data/sr_legacy_food_nutrition_facts.json");
-
 const recommendedDailyIntakes = require("../../data/rdi.json");
+interface Food {
```
Comment (hhimanshu) [inline src/generators/DetailsPageGenerator.ts:13]: This interface is duplicate from `page.d.ts`. Consider removing this one and use from there.

File: src/generators/DetailsPageGenerator.ts [hunk 2]
```diff
@@ -39,15 +49,37 @@ export const createDetailPage = (createPage: any) => {
...
 };
 
-const createNutritionTable = ({ createPageFunction, foods, indexFileName }: any) => {
+const createNutritionTable = ({ createPageFunction, foods }: any) => {
   let ffSearchIndex: any = [];
   const template = path.resolve("src/templates/DetailsPage.tsx");
-  foods.forEach((food: any) => {
+  foods.forEach((food: Food) => {
     if (food.name) {
+      function filterNutrient(food: Food, name: string, unit: string): number {
```
Comment (hhimanshu) [inline src/generators/DetailsPageGenerator.ts:57]: You are creating functions (`filterNutrient`, `generateSEOInfo`) for each food. This is not correct. Create them once, use them multiple times by passing data.

Comment (Ayomikuolatunji) [reply src/generators/DetailsPageGenerator.ts:57]: Fixed

File: src/generators/DetailsPageGenerator.ts (hunk 1 above)
Comment (Ayomikuolatunji) [reply src/generators/DetailsPageGenerator.ts:13]: fixed

**Review by hhimanshu — APPROVED** [review]

**Pull request merged**